envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyEgressRoutesFactory.kt

```diff
@@ -113,18 +113,17 @@ class EnvoyEgressRoutesFactory(
         val routeAction = RouteAction.newBuilder()
             .setCluster(routeSpecification.clusterName)
 
-        routeSpecification.settings.timeoutPolicy?.let { timeoutPolicy ->
+        routeSpecification.settings.timeoutPolicy.let { timeoutPolicy ->
```

Comment:
Why? `timeoutPolicy` is nullable

Reply:
Not really
```kotlin
data class DependencySettings(
    val handleInternalRedirect: Boolean = false,
    val timeoutPolicy: Outgoing.TimeoutPolicy = Outgoing.TimeoutPolicy(),
    val rewriteHostHeader: Boolean = false
.....
    data class TimeoutPolicy(
        val idleTimeout: Duration? = null,
        val requestTimeout: Duration? = null
    )
)
```
This will always have a value even when  idleTimeout and requestTimeout are not set.

Reply:
Apologies, I had older codebase when I looked it up.